### ISSUE TITLE
Route constraints & default parameter values

### ISSUE THREAD

**Issue by tomasherceg** [riganti/dotvvm#671]
We are missing multiple features:
* Multiple route constraints
* Default values in route URLs (`{param:contraint1:constraint2=defaultValue}`)

**Comment by exyi** [comment]
>    * Default values in route URLs (`{param:contraint1:constraint2=defaultValue}`)

We have a separate parameter for default values:
```C#
routes.Add("R1, ""something/{param123}", "R1.dothtml", defaultValues: new { param123 = "default" });
```

**Comment by ridercz** [comment]
> We have a separate parameter for default values:

Yes, but inline specification is more concise and is present in MVC/Razor for several versions now and people are used to it. The separate parameter is old-school :)

**Comment by exyi** [comment]
> inline specification is more concise and is present in MVC/Razor for several versions now and people are used to it. The separate parameter is old-school :)

Well, I'm not closing this issue - just point out you can already do that, so the feature is not missing. However, having duplicate API for such a simple thing is also not something we are looking for and we certainly can't get rid of the old-school one.


@tomasherceg Btw, do you have a use case for multiple constraints? There is also a slight problem with that - the constraint is actually a type specifier and provides a [parse function](https://github.com/riganti/dotvvm/blob/master/src/DotVVM.Framework/Routing/IRouteParameterConstraint.cs#L12) that transforms the string into an object (`{x:int}` will be `int` not `string`). When we'd have more of these parameters, the result type could be ambiguous.

**Comment by ridercz** [comment]
The use case for multiple constraints is when you want to specify type and then some other restricting condition, ie. `"product/{id:int:min(1)}"` - the product ID has to be integer greater than zero

**Comment by tomasherceg** [comment]
I have implemented the support for multiple route constraints and default values in routes, but I need to make sure it behaves the same way as in MVC. And I really don't like how the built-in route constraints in DotVVM are implemented, so I plan to do some refactoring.
